Dockerfile

```diff
@@ -1,39 +1,23 @@
-# Taken from https://github.com/docker-library/golang/blob/master/1.6/Dockerfile 
-# to build our golang image with debian testing (stretch).
-FROM buildpack-deps:stretch-scm
-
-# gcc for cgo.
-RUN apt-get update && apt-get install -y --no-install-recommends \
-		g++ \
-		gcc \
-		libc6-dev \
-		make \
-	&& rm -rf /var/lib/apt/lists/*
-
-ENV GOLANG_VERSION 1.7.1
-ENV GOLANG_DOWNLOAD_URL https://golang.org/dl/go$GOLANG_VERSION.linux-amd64.tar.gz
-ENV GOLANG_DOWNLOAD_SHA256 43ad621c9b014cde8db17393dc108378d37bc853aa351a6c74bf6432c1bbd182
-
-RUN curl -fsSL "$GOLANG_DOWNLOAD_URL" -o golang.tar.gz \
-	&& echo "$GOLANG_DOWNLOAD_SHA256  golang.tar.gz" | sha256sum -c - \
-	&& tar -C /usr/local -xzf golang.tar.gz \
-	&& rm golang.tar.gz
-
-ENV GOPATH /go
-ENV PATH $GOPATH/bin:/usr/local/go/bin:$PATH
-
-RUN mkdir -p "$GOPATH/src" "$GOPATH/bin" && chmod -R 777 "$GOPATH"
-WORKDIR $GOPATH
+FROM golang:1.8
 
 # Build node feature discovery and set it as entrypoint.
 ADD . /go/src/github.com/kubernetes-incubator/node-feature-discovery
 
 WORKDIR /go/src/github.com/kubernetes-incubator/node-feature-discovery
 
 ARG NFD_VERSION
-RUN git clone --depth 1 https://github.com/01org/intel-cmt-cat.git
-RUN cd intel-cmt-cat/lib; make install
-RUN cd rdt-discovery; make
+
+RUN case $(dpkg --print-architecture) in \
+        arm64) \
+                echo "skip rdt on Arm64 platform" \
```

Comment:
Nit: s/skip rdt/skipping RDT feature source

Reply:
Hi @balajismaniam ,
1, The extension instructions in Arm64 platform are different with x86. So I will also update the docs in an independent patch.
Here is a extension instruction list in Arm64 which we collected: For your reference: https://docs.google.com/document/d/13Huvi5bTyqhWp2BQtacaFTaKmeimXLT3wze72MbgkVk/edit

2, For multi-arch issues, I will define the labels with new format, and it is only for arm platform.
Add arm64 in the label,  such as nfd-cpuid-arm64-aes. Just like the format in Openstack.
For your reference: https://docs.openstack.org/os-traits/latest/user/index.html

3, RDT doesn't support arm. We have a similar project in ARM, we called it MPAM.
So for Arm platform, RDT will be skipped, and then we will enable MPAM for NFD.

Reply:
@lubinsz: is arm64 AES somehow different from x86 AES? AFAIU, AES support in arm64 just means support for the AES x86 instruction set extension on arm64 platforms. It's not a completely new capability. So, I would advise the label/annotation be simply "nfd-x86-aes" since it's an x86 instruction set extension, not an arm64 instruction set extension.

Reply:
> 1, The extension instructions in Arm64 platform are different with x86. So I will also update the docs in an independent patch.
Here is a extension instruction list in Arm64 which we collected: For your reference: https://docs.google.com/document/d/13Huvi5bTyqhWp2BQtacaFTaKmeimXLT3wze72MbgkVk/edit

Thanks.

> 2, For multi-arch issues, I will define the labels with new format, and it is only for arm platform.
Add arm64 in the label,  such as nfd-cpuid-arm64-aes. Just like the format in Openstack.
For your reference: https://docs.openstack.org/os-traits/latest/user/index.html

See https://github.com/kubernetes-incubator/node-feature-discovery/issues/86#issuecomment-353184147. Adding the architecture as a part of the label will be redundant.

Reply:
@balajismaniam @jaypipes
I see.
Currently, the architecture will not be a part of the label for arm64 platform